src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerNegativeTests.cs

```diff
@@ -133,12 +165,536 @@ public async Task CreateInvitation_AsViewer_ShouldReturnForbidden_WhenPermission
+                TestFactory.PlantWithAccess,
+                38934,
+                _sigurdSigner.Id,
+                _sigurdSigner.RowVersion,
+                HttpStatusCode.BadRequest);
+        #endregion
+
+        #region Complete
+        [TestMethod]
+        public async Task CompletePunchOut_AsAnonymous_ShouldReturnUnauthorized()
+            => await InvitationsControllerTestsHelper.CompletePunchOutAsync(
+                UserType.Anonymous,
+                TestFactory.PlantWithoutAccess,
+                9999,
+                new CompletePunchOutDto(),
+                HttpStatusCode.Unauthorized);
+
+        [TestMethod]
+        public async Task CompletePunchOut_AsHacker_ShouldReturnBadRequest_WhenUnknownPlant()
+            => await InvitationsControllerTestsHelper.CompletePunchOutAsync(
+                UserType.Hacker,
+                TestFactory.UnknownPlant,
+                9999,
+                new CompletePunchOutDto(),
+                HttpStatusCode.BadRequest,
+                "is not a valid plant");
+
+        [TestMethod]
+        public async Task CompletePunchOut_AsPlanner_ShouldReturnBadRequest_WhenUnknownPlant()
+            => await InvitationsControllerTestsHelper.CompletePunchOutAsync(
+                UserType.Planner,
+                TestFactory.UnknownPlant,
+                9999,
+                new CompletePunchOutDto(),
+                HttpStatusCode.BadRequest,
+                "is not a valid plant");
+
+        [TestMethod]
+        public async Task CompletePunchOut_AsHacker_ShouldReturnForbidden_WhenPermissionMissing()
+            => await InvitationsControllerTestsHelper.CompletePunchOutAsync(
+                UserType.Hacker,
+                TestFactory.PlantWithAccess,
+                9999,
+                new CompletePunchOutDto(),
+                HttpStatusCode.Forbidden);
+
+        [TestMethod]
+        public async Task CompletePunchOut_AsPlanner_ShouldReturnBadRequest_WhenUnknownInvitationId()
+        {
+            var validInvitation = await InvitationsControllerTestsHelper.GetInvitationAsync(
+                UserType.Viewer,
+                TestFactory.PlantWithAccess,
+                InitialInvitationId);
+            var validParticipantForCompleting = _participantsForSigning
+                .Single(p => p.Organization == Organization.Contractor).Person;
+
+            await InvitationsControllerTestsHelper.CompletePunchOutAsync(
+                UserType.Planner,
+                TestFactory.PlantWithAccess,
+                38934,
+                new CompletePunchOutDto()
+                {
+                    InvitationRowVersion = validInvitation.RowVersion,
+                    ParticipantRowVersion = validParticipantForCompleting.RowVersion,
+                    Participants = new List<ParticipantToChangeDto>()
+                },
+                HttpStatusCode.BadRequest);
+        }
+        #endregion
+
+        #region Accept
```

Comment:
I think more negative tests regarding permissions should be added:
* Accept should fail if planner (after removing SIGN from planners permissions list)
* Accept should fail if signer (after removing WRITE from signers permissions list)